src/traces/sankey/convert-to-d3-sankey.js

```diff
@@ -62,7 +100,7 @@ module.exports = function(trace) {
                 label: l,
                 color: hasNodeColorArray ? nodeSpec.color[i] : nodeSpec.color
             });
-        } else removedNodes = true;
+        } else removedNodes = false;
```

Comment:
`d3-sankey` already does that downstrea (ie. removing unlinked nodes) and it's useful to keep if you want to render something associated to it. For example, I use it with d3 to render and animate *ghost* nodes on grouping.